develop/tutorials/articles/03-developing-a-web-application/04-writing-web-application-liferay/01-writing-a-web-application.markdown

```diff
@@ -0,0 +1,635 @@
+*component* is an implementation of an interface. In Liferay, a component is
+typically a
+[Declarative Services](http://wiki.osgi.org/wiki/Declarative_Services)
+component, which means that it's a Java class that's decorated with the
+`org.osgi.service.component.annotations.Component` annotation. Please see the [Understanding Liferay's Module Framework](/develop/tutorials/-/knowledge_base/7-0/understanding-liferays-module-framework)
+tutorial for more information on Liferay modules, components, and Liferay's
+module framework. In this learning path, you're creating a single module called
+`com.liferay.docs.guestbook`. In this module, you're creating a single
+component: the `GuestbookPortlet` class which implements the `Portlet`
+interface.
+
+## What is a Portlet?
+
+When you access a web site, you interact with an application. That application
+might be simple: it might only show you one piece of information, such as an
+article. The application might be complex: you could be doing your taxes
+online, inputting lots of data into an application that calculates whether you
+owe or are due a refund. These applications run on a *platform* that provides
+application developers the building blocks they need to make applications.
+
+![Figure x: Many Liferay applications can run at the same time on the same page.](../../images/portlet-applications.png)
+
+Liferay Portal is a platform, and its applications are called *portlets*. One
+difference between a Liferay application and an application built on another
+platform is that Liferay Portal can serve many applications on the same page at
+the same time. Usually, a web application takes up the entire page. If you
+want, you can do this with Liferay as well. But Liferay has the added benefit
+of being able to run many applications on the same page. Liferay's framework
+takes this into account at every step. For example, features like
+platform-generated URLs exist to support Liferay's ability to serve multiple
+applications on the same page.
+
+Even though all you've done is generate it, the Guestbook project that you
+created in the previous section is ready to be built and deployed to Liferay.
+Make sure your Liferay 7 instance is running. To start it from a Liferay
+workspace, use this command:
+
+    blade server start
+
+To automatically deploy your application using Blade Tools, you first need to
+install the remote agent. To install it, use this command:
+
+    blade agent install
+
+After the blade agent has been installed, run this command from your project:
+
+    blade deploy
+
+The command above both builds and deploys your project.
+
+Next, check that your application is available in Liferay. Open a browser,
+navigate to your portal ([http://localhost:8080](http://localhost:8080) by
+default), and add your application to a page. To add an application to a page,
+click on the *Add* button (it looks like a plus sign) and then on
```

Comment:
You can embed the Add button logo by referring to it (it's in the images folder) within the text.